src/main/scala/com/fulcrumgenomics/sopt/cmdline/CommandLineProgramParser.scala

```diff
@@ -221,25 +240,47 @@ class CommandLineProgramParser[T](val targetClass: Class[T], val includeSpecialA
       case None =>
     }
 
-    // filter on common and partition on optional
-    val (required, optional) = argumentLookup.view
+    // Get all the command line arguments in the order they were declared
+    val arguments: Seq[ClpArgument] = argumentLookup.view
       .filterNot { _.hidden }
       .filter { withSpecial || !_.isSpecial }
-      .partition { !_.optional }
 
-    if (required.nonEmpty) {
-      builder.append(KRED(s"\n${KBLDRED(targetName)} ${KRED(RequiredArguments)}\n"))
-      builder.append(lineBreak(color=true))
-      new ClpArgumentLookup(required:_*).ordered.foreach { arg =>
-        ClpArgumentDefinitionPrinting.printArgumentDefinitionUsage(builder, arg, argumentLookup)
+    // Get all the group names in the order they were first declared in a command line argument
+    val groupNames = mutable.LinkedHashSet[String]()
+    arguments.foreach { arg => groupNames.add(arg.groupName.getOrElse("").capitalize)}
```

Comment:
How do you feel about removing `.capitalize`?  I tend to think that strings for display shouldn't get messed with.  Thought I guess this only toUppers the first char, so not so bad.

Reply:
I want it to capitalize so I don't have to remember in my code.